homeassistant/components/roomba/irobot_base.py

```diff
@@ -182,6 +183,10 @@ def extra_state_attributes(self):
         # Set legacy status to avoid break changes
         state_attrs[ATTR_STATUS] = self.vacuum.current_state
 
+        lastCommand = state.get("lastCommand")
+        if lastCommand is not None:
+          state_attrs['lastCommand'] = json.dumps(lastCommand)
```

Comment:
We generally avoid the use of state attributes as much as possible these days. What would be the use case to add this? (Also, storing a JSON dump in a attribute isn't really end-user consumable output).

Reply:
My particular use case is to allow support for room and zone level cleaning within HA.

In order to do this you first have to setup the rooms and zones withing the iRobot Roomba home app.

Internally the iRobot Roomba home app assigns a numeric values to each room/zone.   The problem is it doesn't expose these values. 

The current method to get these is to either use the dorita980 or rest980 (based on dorita980) APIs and retrieve the lastCommand.   FWIW, the HA Roomba integration is based on these same APIs.
Once you have the lastCommand you use that info to call the vacuum.send_command service in an automation.

Current steps to obtain this:
1. Disable the HA Roomba integration
2. Start a docker image of rest980
3. Repeat (In my case 8 rooms and multiple zones)
    1. In the Roomba app, initiate a individual room/zone clean.
    2. Retrieve lastCommand from rest980
4. Stop docker image
5. Enable the HA Roomba integration
6. Update your automations.

With this update you can eliminate steps 1, 2, 4 and 5.   In addition to eliminating of the docker images.

I used json because it is the same format required for the vacuum.send_command service call.

Reply:
Storing this data in the state machine is going to write it to the database every time there is a new command.  That could get large quite quickly.

That seems like it would be better added as logger messages so you could enable debug logging and get the data.

Reply:
Leaving for anyone interested.

I enabled debug for the HA Roomba integration as suggested:   
logs:
    homeassistant.components.roomba: debug

Then running from a terminal window:
tail -f home-assistant.log | grep lastCommand

I got the data I was looking for when issuing commands.
2022-08-21 09:09:42.938 DEBUG (Thread-3) [homeassistant.components.roomba.irobot_base] Got new state from the vacuum: {'state': OrderedDict([('reported', OrderedDict([('lastCommand', OrderedDict([('command', 'find'), ('initiator', 'localApp'), ('time', 1661087382), ('pmap_id', None), ('regions', None), ('ordered', None), ('user_pmapv_id', None), ('params', None)]))]))])}